test/integration/associations/belongs-to.test.js

```diff
@@ -24,330 +22,218 @@ describe(Support.getTestDialectTeaser('BelongsTo'), function() {
-            expect(result[tasks[2].id]).to.be.undefined;
-          });
-        });
+        return this.sequelize.sync({force: true}).then(() => Sequelize.Promise.join(
+          Task.create({
+            id: 1,
+            user: {id: 1}
+          }, {
+            include: [Task.User]
+          }),
+          Task.create({
+            id: 2,
+            user: {id: 2}
+          }, {
+            include: [Task.User]
+          }),
+          Task.create({
+            id: 3
+          })
+        )).then(tasks => Task.User.get(tasks).then(result => {
+          expect(result[tasks[0].id].id).to.equal(tasks[0].user.id);
+          expect(result[tasks[1].id].id).to.equal(tasks[1].user.id);
+          expect(result[tasks[2].id]).to.be.undefined;
+        }));
       });
     });
   });
 
-  describe('getAssociation', function() {
+  describe('getAssociation', () => {
 
     if (current.dialect.supports.transactions) {
       it('supports transactions', function() {
-        return Support.prepareTransactionTest(this.sequelize).then(function (sequelize) {
-          var User = sequelize.define('User', { username: Support.Sequelize.STRING })
-            , Group = sequelize.define('Group', { name: Support.Sequelize.STRING });
+        return Support.prepareTransactionTest(this.sequelize).then(sequelize => {
+          const User = sequelize.define('User', { username: Support.Sequelize.STRING }), Group = sequelize.define('Group', { name: Support.Sequelize.STRING });
 
           Group.belongsTo(User);
 
-          return sequelize.sync({ force: true }).then(function() {
-            return User.create({ username: 'foo' }).then(function(user) {
-              return Group.create({ name: 'bar' }).then(function(group) {
-                return sequelize.transaction().then(function(t) {
-                  return group.setUser(user, { transaction: t }).then(function() {
-                    return Group.all().then(function(groups) {
-                      return groups[0].getUser().then(function(associatedUser) {
-                        expect(associatedUser).to.be.null;
-                        return Group.all({ transaction: t }).then(function(groups) {
-                          return groups[0].getUser({ transaction: t }).then(function(associatedUser) {
-                            expect(associatedUser).to.be.not.null;
-                            return t.rollback();
-                          });
-                        });
-                      });
-                    });
-                  });
-                });
-              });
-            });
-          });
+          return sequelize.sync({ force: true }).then(() => User.create({ username: 'foo' }).then(user => Group.create({ name: 'bar' }).then(group => sequelize.transaction().then(t => group.setUser(user, { transaction: t }).then(() => Group.all().then(groups => groups[0].getUser().then(associatedUser => {
```

Comment:
Even though its all just setup, I'm not sure I like having it all on a single line